### PR TITLE
DEV: Updates to api docs schema validation

### PR DESCRIPTION
**Pull request by oblakeerickson**
- Read in schemas from actual json files instead of a ruby hash. This is
helpful because we will be automatically generating .json schema files
from json responses and don't want to manually write ruby hash schema
files.

- Create a helper method for rspec schema validation tests to dry up code

### CODE REVIEW

**Comment by oblakeerickson** [comment]
Follow up to: https://github.com/discourse/discourse/pull/11721

File: spec/requests/api/tags_spec.rb [hunk 1]
```diff
@@ -60,17 +60,17 @@
     post 'Creates a tag group' do
       tags 'Tags'
       consumes 'application/json'
-      expected_request_schema = SpecSchemas::TagGroupCreateRequest.new
+      expected_request_schema = SpecSchemas::SpecLoader.new('tag_group_create_request').load
```
Comment (martin-brennan) [inline spec/requests/api/tags_spec.rb:63]: Another opportunity for a helper method -- just sayin' ;). Something like:

```ruby
expected_request_schema = load_spec_schema('tag_group_create_request')
```

**Review by martin-brennan — APPROVED** [review]
Nice, it's good that we will be able to generate the JSON files from actual responses!

File: spec/requests/api/tags_spec.rb (hunk 1 above)
Comment (oblakeerickson) [reply spec/requests/api/tags_spec.rb:63]: Added a helper in https://github.com/discourse/discourse/pull/11849

**Pull request merged**